crates/ide-assists/src/handlers/convert_let_else_to_match.rs

```diff
@@ -0,0 +1,497 @@
+use hir::Semantics;
+use ide_db::RootDatabase;
+use syntax::ast::{edit::AstNodeEdit, AstNode, HasName, LetStmt, Name, Pat};
+use syntax::T;
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+/// Gets a list of binders in a pattern, and whether they are mut.
+fn binders_in_pat(
+    acc: &mut Vec<(Name, bool)>,
+    pat: &Pat,
+    sem: &Semantics<RootDatabase>,
+) -> Option<()> {
+    use Pat::*;
+    match pat {
+        IdentPat(p) => {
+            let ident = p.name()?;
+            let ismut = p.ref_token().is_none() && p.mut_token().is_some();
+            // check for const reference
+            if !(p.is_simple_ident() && sem.resolve_bind_pat_to_const(p).is_some()) {
```

Comment:
```suggestion
            if sem.resolve_bind_pat_to_const(p).is_none()) {
```
This should be enough, if there is `ref` or `mut` keywords on there while it still resolves to a const its a user error anyways since you aren't allowed to use them when the ident resolves to a constant.